tests/test_solver.py

```diff
@@ -54,8 +54,56 @@ def test_FISTA():
     
     stepsize = 0.99/(sp.linalg.interpolative.estimate_spectral_norm(A)**2)
     x0 = np.zeros(n)
-    sol, _ = FISTA(A, b, x0, proximal = ProximalL1, stepsize = stepsize, maxit = 100, abstol=1e-12, adaptive = True).solve()
+    sol, _ = FISTA(A, b, ProximalL1, x0, stepsize = stepsize, maxit = 100, abstol=1e-12, adaptive = True).solve()
 
     ref_sol = np.array([-1.83273787e-03, -1.72094582e-13,  0.0, -3.35835639e-01, -1.27795593e-01])
     # Compare
-    assert np.allclose(sol, ref_sol, atol=1e-4)
\ No newline at end of file
+    assert np.allclose(sol, ref_sol, atol=1e-4)
+
+def test_ADMM_matrix_form():
+    # Parameters
+    rng = np.random.default_rng(seed = 42)
+    m, n = 10, 5
+    A = rng.standard_normal((m, n))
+    b = rng.standard_normal(m)
+    
+    k = 4
+    L = rng.standard_normal((k, n))
+
+    stepsize = 0.99/(sp.linalg.interpolative.estimate_spectral_norm(A)**2)
+    x0 = np.zeros(n)
+    sol, _ = ADMM(A, b, [(ProximalL1, np.eye(n)), (lambda z, _ : ProjectNonnegative(z), L)],
+                   x0, 10, maxit = 100, adaptive = True).solve()
+
+    ref_sol = np.array([-3.99513417e-03, -1.32339656e-01, -4.52822633e-02, -7.44973888e-02, -3.35005208e-11])
+    # Compare
+    assert np.allclose(sol, ref_sol, atol=1e-4)
+
+
+def test_ADMM_function_form():
+    # Parameters
+    rng = np.random.default_rng(seed = 42)
+    m, n = 10, 5
+    A = rng.standard_normal((m, n))
+    def A_fun(x, flag):
+        if flag == 1:
+            return A@x
+        if flag == 2:
+            return A.T@x
+        
+    b = rng.standard_normal(m)
+    
+    k = 4
+    L = rng.standard_normal((k, n))
+
+    stepsize = 0.99/(sp.linalg.interpolative.estimate_spectral_norm(A)**2)
+    x0 = np.zeros(n)
+    sol, _ = ADMM(A_fun, b, [(ProximalL1, np.eye(n)), (lambda z, _ : ProjectNonnegative(z), L)],
+                   x0, 10, maxit = 100, adaptive = True).solve()
+
+    print(sol)
+    ref_sol = np.array([-3.99513417e-03, -1.32339656e-01, -4.52822633e-02, -7.44973888e-02, -3.35005208e-11])
+    # Compare
+    assert np.allclose(sol, ref_sol, atol=1e-4)
+
+test_ADMM_function_form()
```

Comment:
There seem to be a duplicate method here? Not sure if that is the cause for the unit tests failing. It looked like they began failing when the test "test_ADMM_function_form" was added.